asio/include/asio/detail/config.hpp

```diff
@@ -559,6 +560,32 @@
 # endif // !defined(ASIO_RVALUE_REF_QUAL)
 #endif // defined(ASIO_HAS_REF_QUALIFIED_FUNCTIONS)
 
+// Support for the alignof operator.
+#if !defined(ASIO_HAS_ALIGNOF)
+# if !defined(ASIO_DISABLE_ALIGNOF)
+#  if (__cplusplus >= 201103)
+#   define ASIO_HAS_ALIGNOF 1
+#  endif // (__cplusplus >= 201103)
+# endif // !defined(ASIO_DISABLE_ALIGNOF)
+#endif // !defined(ASIO_HAS_ALIGNOF)
+
+#if defined(ASIO_HAS_ALIGNOF)
+# define ASIO_ALIGNOF(T) alignof(T)
+# define ASIO_DEFAULT_ALIGN alignof(std::max_align_t)
```

Comment:
Note that `max_align_t` gives the maximum alignment of scalar types and not the maximum possible alignment. Dynamic allocations should already be aligned to that value, so this shouldn't change the alignment of non over-aligned types.